tests/smartswitch/common/device_utils_dpu.py

```diff
@@ -37,29 +37,31 @@ def check_smartswitch_and_dark_mode(duthosts,
     """
 
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    output = duthost.os_version
+    os_version = float(re.search(r'([1-9]{1}\d*)(\.\d{0,2})', output).group())
 
-    if not duthost.facts["DPUS"] and \
-            parse_version(duthost.os_version) <= parse_version("202405"):
+    if "DPUS" not in duthost.facts and \
+            os_version <= float("202405"):
         pytest.skip("Test is not supported for this testbed and os version")
 
-    darkmode = is_dark_mode_enabled(duthost, platform_api_conn)
+    darkmode = is_dark_mode_enabled(duthost, platform_api_conn,
+                                    num_dpu_modules)
 
     if darkmode:
         dpu_power_on(duthost, platform_api_conn)
```

Comment:
dpu_power_on also takes num_dpu_modules as argument.

Reply:
Changed it.